### ISSUE TITLE
Provide more helpful error when remote name does not exist

### ISSUE THREAD

**Issue by kguidonimartins** [git-lfs/git-lfs#4194]
Folks, thanks for your work on this. This tool has made my life easier in the last years. However, in the last weeks, I cannot push my files. Can you help with this?

I have received the following message:

```bash
batch request: missing protocol: "origin.git/info/lfs"
error: failed to push some refs to 'https://github.com/kguidonimartins/check-trait-sampling-location-bien.git'
```

#### git config -l:

```bash
$ git config -l

user.email=kguidonimartins@gmail.com
user.name=Karlo Guidoni Martins
gui.editor=nvim
alias.lg=log --color --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit
alias.ctags=!.git/hooks/ctags
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
github.user=kguidonimartins
core.filemode=false
core.pager=diff-so-fancy | less --tabs=4 -RFX
core.excludesfile=~/.config/git/ignore
hub.protocol=https
init.templatedir=~/.config/git/template
credential.helper=store --file ~/.config/git/credentials
lfs.locksverify=false
travis.slug=kguidonimartins/virtual-biotic-pollination-flow
```

#### git lfs env

```bash
$ git lfs env

git-lfs/2.11.0 (GitHub; linux amd64; go 1.14.4)
git version 2.27.0

Endpoint=origin.git/info/lfs (auth=none)
LocalWorkingDir=/home/karlo/Downloads/check-trait-sampling-location-bien
LocalGitDir=/home/karlo/Downloads/check-trait-sampling-location-bien/.git
LocalGitStorageDir=/home/karlo/Downloads/check-trait-sampling-location-bien/.git
LocalMediaDir=/home/karlo/Downloads/check-trait-sampling-location-bien/.git/lfs/objects
LocalReferenceDirs=
TempDir=/home/karlo/Downloads/check-trait-sampling-location-bien/.git/lfs/tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=/home/karlo/Downloads/check-trait-sampling-location-bien/.git/lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file
UploadTransfers=basic,lfs-standalone-file
FORGIT_GI_REPO_LOCAL=/home/karlo/.forgit/gi/repos/dvcs/gitignore
FORGIT_GI_REPO_REMOTE=https://github.com/dvcs/gitignore
FORGIT_GI_TEMPLATES=/home/karlo/.forgit/gi/repos/dvcs/gitignore/templates
GIT_CONFIG=/home/karlo/.config/git/config
GIT_EXEC_PATH=/usr/lib/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

Output using recommendations of previous issues founded here:

```bash
$ GIT_TRACE=1 git push origin main

09:24:56.298674 git.c:442               trace: built-in: git push -u origin main
09:24:56.300605 run-command.c:663       trace: run_command: GIT_DIR=.git git-remote-https origin https://github.com/kguidonimartins/check-trait-sampling-location-bien.git
09:25:01.978294 run-command.c:663       trace: run_command: 'git credential-store --file ~/.config/git/credentials get'
09:25:01.991015 git.c:702               trace: exec: git-credential-store --file /home/karlo/.config/git/credentials get
09:25:01.991151 run-command.c:663       trace: run_command: git-credential-store --file /home/karlo/.config/git/credentials get
09:25:02.616796 run-command.c:663       trace: run_command: 'git credential-store --file ~/.config/git/credentials store'
09:25:02.632090 git.c:702               trace: exec: git-credential-store --file /home/karlo/.config/git/credentials store
09:25:02.632181 run-command.c:663       trace: run_command: git-credential-store --file /home/karlo/.config/git/credentials store
09:25:02.636976 run-command.c:663       trace: run_command: .git/hooks/pre-push origin https://github.com/kguidonimartins/check-trait-sampling-location-bien.git
09:25:02.645758 git.c:702               trace: exec: git-lfs pre-push origin https://github.com/kguidonimartins/check-trait-sampling-location-bien.git
09:25:02.645866 run-command.c:663       trace: run_command: git-lfs pre-push origin https://github.com/kguidonimartins/check-trait-sampling-location-bien.git
09:25:02.662820 trace git-lfs: exec: git 'version'
09:25:02.669802 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
09:25:02.674654 trace git-lfs: exec: git 'config' '-l'
09:25:02.683970 trace git-lfs: pre-push: refs/heads/main 66dede75b285d2a47db481a6fd5aafbfa81584a1 refs/heads/main 0000000000000000000000000000000000000000
09:25:03.295786 trace git-lfs: tq: running as batched queue, batch size of 100
09:25:03.296168 trace git-lfs: run_command: git rev-list --objects --ignore-missing --not --remotes=origin --stdin --
09:25:03.304542 trace git-lfs: tq: sending batch of size 1
batch request: missing protocol: "origin.git/info/lfs"
error: failed to push some refs to 'https://github.com/kguidonimartins/check-trait-sampling-location-bien.git'
```

**Comment by bk2204** [comment]
Hey,

I think your issue is that you don't have a remote `origin` set up.  If you run `git remote -v`, it will likely show no output.  So when you run `git push origin main`, you're trying to push to a directory named `origin`, except Git LFS knows full well that such a directory doesn't exist, and so doesn't consider it a local directory.  Git would probably give you a slightly more helpful error message if Git LFS weren't involved.

The solution to this, of course, is just to add a remote with `git remote add origin URL`, where `URL` is wherever you want to push things.  I'll see if we can get a better error for that case; we didn't consider that case when adding local file support and so didn't consider a more helpful error message.

**Comment by kguidonimartins** [comment]
Hey, thanks for the reply. 

My repository has the origin. This error have occurred in other repos too.

```bash
$ git remote -v 
                                                                                                                                                                 
origin  https://github.com/kguidonimartins/check-trait-sampling-location-bien.git (fetch)
origin  https://github.com/kguidonimartins/check-trait-sampling-location-bien.git (push)
```

**Comment by bk2204** [comment]
Hey,

I don't see that in the `git config -l` output.  Did you strip it out, or are you configuring the remotes differently somehow?

**Comment by kguidonimartins** [comment]
Hey, 

That was the full output of `git config -l`. I'm configuring the remotes as usual, using `git remote add origin URL`.

**Comment by bk2204** [comment]
Do you have an `.lfsconfig` file?  If so, can you include the contents as a code block?

**Comment by kguidonimartins** [comment]
> Do you have an `.lfsconfig` file? If so, can you include the contents as a code block?

Yup. I did include `.lfsconfig` in this [repo](https://github.com/kguidonimartins/check-trait-sampling-location-bien.git) after report the issue.

```
[remote "origin"]
        lfsurl = https://github.com/kguidonimartins/check-trait-sampling-location-bien.git
```

 So, the error message changed:

```
batch response: Authorization error: https://github.com/kguidonimartins/check-trait-sampling-location-bien.git/objects/batch
Check that you have proper access to the repository
Uploading LFS objects:   0% (0/1), 0 B | 0 B/s, done.
error: failed to push some refs to 'https://github.com/kguidonimartins/check-trait-sampling-location-bien.git'
``` 

Other repos, which does not have `.lfsconfig`, the error message is the same reported early. For example:

```
batch request: missing protocol: "origin.git/info/lfs"
error: failed to push some refs to 'https://github.com/kguidonimartins/dados-mapa-interativo.git'
```

**Comment by bk2204** [comment]
My concern is that if you've run `git config -l` in the repository, I should see an entry for `remote.origin.url`, but I don't see that.  If that value is missing, I have trouble understanding how Git thinks you have that remote set, and I'm concerned that other settings that are relevant to fixing the problem are also missing.  You said that you're seeing this for multiple repositories, so there's clearly a relevant difference between your setup and mine (where this works), so I'm interested in finding out where those settings have gone, since I think they're relevant.

As for your new issue, I think if you want to specify the `lfsurl` setting, you also need to tack on `/info/lfs` to the end of the URL you're using.

**Comment by kguidonimartins** [comment]
> As for your new issue, I think if you want to specify the lfsurl setting, you also need to tack on /info/lfs to the end of the URL you're using.

Indeed, adding `/info/lfs` solved the problem. I set the `.lfsconfig` file to the other repositories and the issue was also resolved. The source of the problem is somewhat strange though. Anyway, having the `.lfsconfig` file at the root of the project makes it clear that the repository uses `git lfs`. I liked this option.

Thank you so much for your time!

I'm closing this issue.
